Add admin member request feature for communities

jrainville opened the pull request:
Fixes #4442

The issue itself could have been solved by just commenting stuff, but instead I implemented the feature on the fron-end.

Now, community admins can receive membership requests for "on-request" communities.
In the modal, they can accept of decline those requests.

To test this, I used a test mobile account. I sent the community link in a public chat, and opened the link in mobile. If the community data is not fetched, you can edit the community (just change the description) on desktop and the community data should propagate.
From there, just Request access on mobile. The Membership request will appear on desktop.

Depends on https://github.com/status-im/status-lib/pull/164

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d61c7b20 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-4465/1/) | 2022-01-14 20:48:38 | ~16 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220114-203155-d61c7b-pr4465.dmg) |

saledjenic reviewed (changes requested):
@jrainville changes you did look ok to me. But I have some more general concerns... somehow seems that responsibilities are mixed a bit among `communitiesModule` and chat section module which represents a community. And because of that we may run into many other issues later or that may start even now (as it's an inconsistency we have at the image I added below (don't look at the members count which is also not correct (should be 2 members) on that image, but members count we have in that "manage community popup" is not the same as members count we have in "community info title bar", but they ought to be the same if they are fetching the data form the same source which should be chat section module which represents that community)). 

So how it should be... `communitiesModule` is identified as a separate module just to provide necessary convenience when we're joining/creating community, nothing else (that's the only purpose of that module, don't look how it was in the old code). So it should be used only when user access "Communities" popup via click on `Chat section > Plus button > Communities action from that menu`. A responsibilities of that module (`communitiesModule`) are to list all available communities, provide necessary info about them, initiate a creation of a new community. In all other cases (after once a community has been created) all other actions/features should work with appropriate chat section module which represents active community section and those actions are responsibilities of that module.

We can discuss this further if you want in private chat/call.

This is an image I mentioned above:

![communities](https://user-images.githubusercontent.com/86303051/149678254-42eca411-0b4d-4535-bbaa-b0885e310a15.png)

jrainville commented on src/app/modules/shared_models/active_section.nim:
```diff
@@ -127,4 +127,13 @@ QtObject:
+
+  proc pendingRequestsToJoin(self: ActiveSection): QVariant {.slot.} =
+    if (self.item.id == ""):
+      # FIXME (Jo) I don't know why but the Item is sometimes empty and doing anything here crashes the app
```
It's pretty easy. To reproduce, you just need to comment those lines and have a joinedCommunity, for example, create a new community, close the app and reopen

saledjenic replied:
@jrainville yes, I managed to reproduce a crash when I removed that "if" condition, thanks for the tip.

What I see... the issue is somehow in relation to my concerns added [here](https://github.com/status-im/status-desktop/pull/4465#pullrequestreview-853734668), there are places on the qml side where you're accessing `pendingRequestsToJoin` and `members` properties as a part of `communitiesModule` (via `observedCommunity`) when we load qml, but you don't have `observedCommunity` set in that moment since it's set from the qml side onlly, on demand via `setObservedCommunity` proc, and that's why you faced a crash.

For the difference how is that working in `mainModule` in `load` we have:
```
# Set active section on app start
self.setActiveSection(activeSection)
```
what ensured that active section is set before qml is loaded, and from that point on we are able to change active section from the qml side.

The thing here is a "mixture" I mentioned above, cause you cannot set "observedCommunity" community before we actually need to do that (so that cannot be set on the app load), but actually `CommunityProfilePopup` and components beneath should operate with appropriate chat section module which represents active community section.

jrainville replied:
I added a new PR on top of this one that removes the use of observedCommunity. Sadly, it didn't fix the crash. You can see the PR here: https://github.com/status-im/status-desktop/pull/4488

Let me know @saledjenic if you have other ideas of how to fix that.

jrainville commented:
@saledjenic I rebased and fixed the two line comments you pointed out.

> the members count which is also not correct (should be 2 members) on that image, but members count we have in that "manage community popup" is not the same as members count we have in "community info title bar"

This was fixed in the other PR https://github.com/status-im/status-desktop/pull/4412
Member list is not the same as membership request list

> communitiesModule is identified as a separate module just to provide necessary convenience when we're joining/creating community, nothing else [...]

I understand your point. It makes sense, especially if we want the backend to match perfectly with the front-end. I created this issue https://github.com/status-im/status-desktop/issues/4473 to work on it later, since it's not that much of a priority vs making communities work

saledjenic commented on src/app_service/service/community/service.nim:
```diff
@@ -98,6 +102,16 @@ QtObject:
       error "error: ", errDesription
       return
 
+    self.events.on(SignalType.Message.event) do(e:Args):
```
@jrainville I just commit a minor thing here instead of writing a comment, just removed `*` to make `doConnect` a private method. Hope that you don't mind.

saledjenic commented:
> Member list is not the same as membership request list

@jrainville yes, I know that, but look at the screenshot I posted above, top left arrow point to the number of community members which is `1 member`, but in the popup "Members" row, left from the number of new membership request, there is `0` members, and I think (but maybe I am wrong, maybe that number represents something else) that number should display `1` if that's the number of members that community currently has?

jrainville commented:
Alright, I reased and merged my other PR, so your issues should all be fixed @saledjenic . You can review again.

saledjenic reviewed (approved):
@jrainville I am approving this since changes done here look good. Cause here is a lot of changes maybe we should merge it and work on the following issues through a new tickets (but if you want you can do it here):

- when we create new private community and one ask to join, after request is approved by the community admin, joined user is not added to the users list on the right:

![commissue001](https://user-images.githubusercontent.com/86303051/150494327-e5dc32ba-6f10-4169-b4e3-91bc521d1cb6.png)

- after app restart we have a bit different issue that logged in user is added twice:

![commissue002](https://user-images.githubusercontent.com/86303051/150494448-e0c358f2-fd19-44b9-aca8-e15e1bf08403.png)

jrainville commented:
@saledjenic good find. I'll fix it in another PR, especially since I'll also need this PR to be merged for it to work https://github.com/status-im/status-desktop/pull/4522

jrainville commented:
Issue created here https://github.com/status-im/status-desktop/issues/4523

jrainville commented:
rebased and ready to test @abhi-status
